### PR TITLE
add README for State Query API

### PR DESCRIPTION
**Pull request by dmitsh**
# Description

Added README for State Query API

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [x] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#1863

### CODE REVIEW

**Comment by dmitsh** [comment]
This PR takes dependency on https://github.com/dapr/docs/pull/1874

File: state/Readme.md [hunk 1]
```diff
@@ -51,3 +51,146 @@ type TransactionalStore interface {
...
+The last part is to implement `Querier` interface in the component:
+```go
+type Querier interface {
+	Query(req *QueryRequest) (*QueryRespond, error)
+}
+```
+
+A sample implementation might look like that:
+```go
+func (m *MyComponent) Query(req *state.QueryRequest) (*state.QueryRespond, error) {
```
Comment (pkedy) [inline state/Readme.md:176]: Should `QueryRespond` be `QueryResponse`?

File: state/Readme.md [hunk 2]
```diff
@@ -51,3 +51,146 @@ type TransactionalStore interface {
...
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	query := &Query{} // Query implements Visitor interface
+	qbuilder := state.NewQueryBuilder(query)
+	if err := qbuilder.BuildQuery(&req.Query); err != nil {
+		return &state.QueryRespond{}, err
```
Comment (pkedy) [inline state/Readme.md:183]: Should this be `return nil, err`?

File: state/Readme.md [hunk 3]
```diff
@@ -51,3 +51,146 @@ type TransactionalStore interface {
...
+	}
+	data, token, err := query.execute(ctx)
+	if err != nil {
+		return &state.QueryRespond{}, err
```
Comment (pkedy) [inline state/Readme.md:187]: Should this be `return nil, err`?

**Review by msfussell — COMMENTED** [review]
@ dmitsh - Whilst you are in this file, can you make the color syntax between the existing state stores interfaces, match the color the the one you added for Query. For example type is in red. Just a consistency aspect :-) Otherwise LGTM

File: state/Readme.md (hunk 2 above)
Comment (dmitsh) [reply state/Readme.md:183]: I was following the examples of Get/Set. They return an empty response. But I agree, it looks weird.

File: state/Readme.md (hunk 3 above)
Comment (dmitsh) [reply state/Readme.md:187]: same as above

**Review by artursouza — APPROVED** [review]

**Pull request merged**